.github/workflows/build.yml

```diff
@@ -0,0 +1,38 @@
+name: build
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
Looks like it could be done with something like [the following](https://github.community/t/setting-default-shell-or-other-step-metadata-conditionally-in-workflows/154055/2), or like [this](https://github.community/t/conditional-matrices/17206/2).

Reply:
The first option looks pretty good

Reply:
I don't know if there's an easy way to build gcc and clang only for ubuntu, it might be easier to split the platforms up into their own configs at this point.